include/dgl/immutable_graph.h

```diff
@@ -889,6 +914,9 @@ class ImmutableGraph: public GraphInterface {
     if (!in_csr_) {
       if (out_csr_) {
         const_cast<ImmutableGraph*>(this)->in_csr_ = out_csr_->Transpose();
+        if (out_csr_->IsSharedMem())
+          LOG(WARNING) << "We just construct an in-CSR from a shared-memory out CSR. "
+                       << "It may dramatically increase memory consumption.";
```

Comment:
What does this mean? Is the `in_csr` also in shared memory? Is there any solution to this warning?

Reply:
I add this warning in case this happens. We could make sure the newly constructed csr is also in shared memory.